goal_src/jak2/levels/common/battle.gc

```diff
@@ -7,3 +7,1875 @@
+               (a2-2 (handle->process (-> a1-5 creature)))
+               )
+          (when (and a2-2 (= a2-2 v1-0))
+            (when (not (logtest? (enemy-flag alert) (-> (the-as nav-enemy a2-2) enemy-flags)))
+              (set! (-> a1-5 creature) (the-as handle #f))
+              (set! a1-5 (the-as battle-spawner #f))
+              )
+            (return a1-5)
+            )
+          )
+        )
+      )
+    )
+  (the-as battle-spawner #f)
+  )
+
+(defmethod battle-method-51 battle ((obj battle) (arg0 battle-spawner) (arg1 symbol))
+  (when (and (logtest? (the-as int (-> obj flags)) 2) (zero? (logand (the-as int (-> obj flags)) 4)))
+    (let ((v1-5 (-> arg0 noticed-attack-time)))
+      (cond
+        ((zero? v1-5)
+         (set! (-> arg0 noticed-attack-time) (-> self clock frame-counter))
+         )
+        (else
+          (if (>= (- (-> self clock frame-counter) v1-5) (the-as time-frame (-> arg0 notice-attack-delay)))
+              (logior! (-> arg0 flags) 1)
+              )
+          )
+        )
+      )
+    )
+  (let ((s4-0 (handle->process (-> arg0 creature))))
+    (when s4-0
+      (when (not (logtest? (new 'static 'handle :process #x20000 :u64 #x20000) (-> (the-as touch-tracker s4-0) target)))
+        (set! (-> arg0 creature) (the-as handle #f))
+        (return #f)
+        )
+      (let ((v1-20 (-> arg0 mode)))
+        (cond
+          ((= v1-20 1)
+           (set! (-> arg0 mode) (the-as uint 0))
+           0
+           )
+          ((zero? v1-20)
+           (when (or (logtest? (new 'static 'handle :process #x40000 :u64 #x40000) (-> (the-as touch-tracker s4-0) target))
+                     (not arg1)
+                     )
+             (cond
+               ((battle-method-46 obj arg0)
+                (if (not (battle-method-43 obj arg0 (the-as touch-tracker s4-0)))
+                    (return #t)
+                    )
+                )
+               ((battle-method-45 obj arg0 (the-as touch-tracker s4-0))
+                (battle-method-42 obj arg0 (the-as touch-tracker s4-0))
+                )
+               )
+             )
+           )
+          )
+        )
+      )
+    )
+  #f
+  )
+
+(defmethod battle-method-46 battle ((obj battle) (arg0 battle-spawner))
+  (when (-> arg0 intro-path)
+    (let ((v1-1 (-> arg0 creature-index))
+          (a0-1 (-> arg0 attack-index))
+          )
+      (or (< v1-1 (-> arg0 ready-index)) (and (< v1-1 a0-1) (logtest? (-> arg0 flags) 1)))
+      )
+    )
+  )
+
+(defmethod battle-method-43 battle ((obj battle) (arg0 battle-spawner) (arg1 touch-tracker))
+  (let ((s3-0 (+ (-> arg0 creature-index) 1))
+        (s4-0 (new 'stack-no-clear 'vector))
+        )
+    (if (and (= s3-0 (-> arg0 attack-index)) (battle-method-25 obj arg0))
+        (return #f)
+        )
+    (get-point-in-path! (-> arg0 intro-path) s4-0 (the float s3-0) 'exact)
+    (let ((s3-1 (-> arg0 mode)))
+      (set! (-> arg0 mode) (the-as uint 2))
+      (set! (-> arg1 target)
```

Comment:
I think this isn't a handle.  No idea what it is though.

Reply:
this should still work, I left a note for this file and we can come back later. It'll be easier once we understand more of this code.